tests/tests_unit/test_api_client.py

```diff
@@ -1009,22 +1011,41 @@ def test_standard_update_fail_5xx(self, api_client_with_token, rsps):
         assert e.value.failed == []
         assert e.value.unknown == [0, "abc"]
 
-    def test_standard_update_fail_missing_and_5xx(self, api_client_with_token, rsps):
+    def test_standard_update_fail_missing_and_5xx(self, api_client_with_token, rsps, monkeypatch):
+        # Note 1: We have two tasks being added to an executor, but that doesnt mean we know the
+        # execution order. Depending on whether the 400 or 500 hits the first or second task,
+        # the following asserts fail (ordering issue). Thus, we use 'matchers.json_params_matcher'
+        # to make sure the responses match the two tasks.
+
+        # Note 2: The matcher function expects request.body to not be gzipped (it just does .decode("utf-8")
+        # which fails, making the matching functions useless.. so we temporarily turn off gzip for this test
+        monkeypatch.setattr(global_config, "disable_gzip", True)
+
         rsps.add(
             rsps.POST,
             BASE_URL + URL_PATH + "/update",
             status=400,
             json={"error": {"message": "Missing ids", "missing": [{"id": 0}]}},
+            match=[matchers.json_params_matcher({"items": [{"update": {}, "id": 0}]})],
```

Comment:
This `matchers` is used for the `rsps` object to figure out which of these two responses to return based on the request? Very elegant solution.